### ISSUE TITLE
MatchingEventTypes enum values don't match AWS documentation

### ISSUE THREAD

**Issue by billduvall** [aws-cloudformation/cfn-lint#3836]
### CloudFormation Lint Version

1.20.0

### What operating system are you using?

Mac

### Describe the bug

The enum list in the following cfn-lint schema doesn't seem to match the AWS documented list of event types.
 https://github.com/aws-cloudformation/cfn-lint/blob/main/src/cfnlint/data/schemas/providers/us_east_1/aws-ses-configurationseteventdestination.json#L85

AWS documentation for AWS::SES::ConfigurationSetEventDestination EventDestination MatchingEventTypes
https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-ses-configurationseteventdestination-eventdestination.html#cfn-ses-configurationseteventdestination-eventdestination-matchingeventtypes
![image](https://github.com/user-attachments/assets/34222925-8222-4089-bbc5-8180d2aca79a)

### Expected behavior

I am expecting the AWS::SES::ConfigurationSetEventDestination resource in 'Reproduction template' to not cause the following errors based on the AWS documentation.

281:13:             E3030     'BOUNCE' is not one of ['bounce', 'click', 'complaint', 'delivery', 'open', 'reject', 'renderingFailure', 'send']
282:13:             E3030     'COMPLAINT' is not one of ['bounce', 'click', 'complaint', 'delivery', 'open', 'reject', 'renderingFailure', 'send']
283:13:             E3030     'DELIVERY' is not one of ['bounce', 'click', 'complaint', 'delivery', 'open', 'reject', 'renderingFailure', 'send']
284:13:             E3030     'DELIVERY_DELAY' is not one of ['bounce', 'click', 'complaint', 'delivery', 'open', 'reject', 'renderingFailure', 'send']
285:13:             E3030     'REJECT' is not one of ['bounce', 'click', 'complaint', 'delivery', 'open', 'reject', 'renderingFailure', 'send']
286:13:             E3030     'SUBSCRIPTION' is not one of ['bounce', 'click', 'complaint', 'delivery', 'open', 'reject', 'renderingFailure', 'send']

### Reproduction template

```yml
Resources:
  FeedbackEventDestination:
    Type: AWS::SES::ConfigurationSetEventDestination
    Properties:
      ConfigurationSetName: !Ref SESConfigurationSet
      EventDestination:
        Name: "FeedbackEventDestination"
        Enabled: true
        MatchingEventTypes:
          - BOUNCE
          - COMPLAINT
          - DELIVERY
          - DELIVERY_DELAY
          - REJECT
          - SUBSCRIPTION
        SnsDestination:
          TopicARN: !Ref SESFeedbackTopic
```

**Comment by billduvall** [comment]
Following the AWS documentation the enum values in the json files have a several discrepancies.

1) `RENDERINGFAILURE` should be `RENDERING_FAILURE`
2) `DELIVERY_DELAY` is missing from enum values
3) `SUBSCRIPTION` is missing from enum values
![image](https://github.com/user-attachments/assets/2ab598a0-de55-48a6-b4b7-169f413a6b95)
